open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/properties/objects/graph/Line.java

```diff
@@ -75,16 +79,48 @@ public Line(Line template) {
         this.setGuid(template.getGuid());
         this.setLineType(template.getLineType());
         this.setAdditionalProperties(template.getAdditionalProperties());
+        this.setLineEnds();
     }
 
     public Line(String name) {
-        this.name = name;
-        this.lineType = LineType.Unknown;
+        this(name,LineType.Unknown);
     }
 
     public Line(String name, LineType lineType) {
         this.name = name;
         this.lineType = lineType;
+        this.setLineEnds();
+    }
+    protected void setLineEnds() {
+        OpenMetadataTypesArchiveAccessor archiveAccessor = OpenMetadataTypesArchiveAccessor.getInstance();
```

Comment:
I am alarmed by the use of this service here...

Reply:
> I am alarmed by the use of this service here...

yes good spot - I guess this implies that we need access to the archive in the client libraries. That is a shame as it saves a lot of code.  I might try to generate some code from the archive so we do not have this dependancy.

Reply:
@davidradl I really dont know how better resolve this problem but it is problem...

Reply:
@immmus I have resolved the problem by hard coding the values in the Line children. The code is ready for a new review now.